std/utf.d

```diff
@@ -3175,6 +3192,29 @@ auto ref byCodeUnit(R)(R r)
     return r;
 }
 
+///
+unittest
+{
+    auto r = "Hello, World!".byCodeUnit();
+    static assert(hasSlicing!(typeof(r)));
+    static assert(isRandomAccessRange!(typeof(r)));
+
+    // contrast with the range capabilities of standard strings
+    auto s = "Hello, World!";
+    static assert(isBidirectionalRange!(typeof(r)));
+
+    static assert(!isRandomAccessRange!(typeof(s)));
+    static assert(!hasSlicing!(typeof(s)));
+}
+
+/// `byCodeUnit` does no Unicode encoding
+unittest
+{
+    auto s = "noe\u0308l".byCodeUnit; // noël using e + combining diaeresis
+    s.popFrontN(3);
+    assert(s.front == 'Ì');
```

Comment:
This is misleading. That equality only holds because 0xCC is implicitly converted from `char` to `dchar`. As a `char` that value is part of a multibyte sequence. As a `dchar` it's the character you put there. The implicit conversion is problematic, but this is not `byCodeUnit`'s fault.

Maybe pop only 2 elements, and show that `front` is `'e'` instead of `'ë'`. And/or use a combined ë (U+00EB) to show that that value cannot be found in the elements of `byCodeUnit`, as it's encoded as multiple code units.